dot/sync/chain_processor.go

```diff
@@ -74,7 +74,7 @@ func (s *chainProcessor) processReadyBlocks() {
 			return
 		}
 
-		if err := s.processBlockData(bd); err != nil {
+		if err := s.processBlockData(*bd); err != nil {
```

Comment:
is there a guarantee that `bd` is not nil? I think you should put a check to ensure `bd` is not nil before dereference it

Reply:
It's kinda obscure, but yes; `bd := s.readyBlocks.pop(s.ctx)` only returns `nil` if the context gets canceled, and there is a

```go
		if s.ctx.Err() != nil {
			return
		}
```

check below. I changed that if check to `bd == nil` to make it more explicit :wink: